sparse/core.py

```diff
@@ -233,10 +233,19 @@ def __sizeof__(self):
     def __getitem__(self, index):
         if not isinstance(index, tuple):
             index = (index,)
-        index = tuple(ind + self.shape[i] if isinstance(ind, numbers.Integral) and ind < 0 else ind
+        if len(index) - index.count(None) > self.ndim:
+            raise IndexError("too many indices for array")
```

Comment:
This will fail on things like `a[1, ...]` where `a` is `(n,)`-shaped. There should be an additional factor of `- index.count(Ellipsis)` here.